src/bio/bio_xstream.c

```diff
@@ -766,6 +789,11 @@ create_bio_bdev(struct bio_xs_context *ctxt, const char *bdev_name,
 	}
 
 	D_INIT_LIST_HEAD(&d_bdev->bb_link);
+	rc = bdev_name2roles(bdev_name);
+	if (rc < 0)
+		goto error;
```

Comment:
so we assume the daos_nvme.conf will always be re-generated after upgrading?

Reply:
I haven't got conclusion from Tom yet,  just assume we will for simple reason.